src/components/stack-chart/Canvas.js

```diff
@@ -149,33 +152,50 @@ class StackChartCanvas extends React.PureComponent<Props> {
         viewportBottom,
       },
     } = this.props;
+    const fastFillStyle = new FastFillStyle(ctx);
 
-    // Ensure the text measurement tool is created, since this is the first time
-    // this class has access to a ctx.
-    if (!this._textMeasurement) {
-      this._textMeasurement = new TextMeasurement(ctx);
-    }
-    const textMeasurement = this._textMeasurement;
+    const { devicePixelRatio } = window;
+
+    // Set the font size before creating a text measurer.
+    ctx.font = `${FONT_SIZE * devicePixelRatio}px sans-serif`;
+    const textMeasurement = new TextMeasurement(ctx);
+
+    const devicePixelsWidth = containerWidth * devicePixelRatio;
+    const devicePixelsHeight = containerHeight * devicePixelRatio;
 
-    ctx.fillStyle = '#ffffff';
-    ctx.fillRect(0, 0, containerWidth, containerHeight);
+    fastFillStyle.set('#ffffff');
+    ctx.fillRect(0, 0, devicePixelsWidth, devicePixelsHeight);
 
     const rangeLength: Milliseconds = rangeEnd - rangeStart;
     const viewportLength: UnitIntervalOfProfileRange =
       viewportRight - viewportLeft;
+    const viewportDevicePixelsTop = viewportTop * devicePixelRatio;
 
     // Convert CssPixels to Stack Depth
     const startDepth = Math.floor(viewportTop / stackFrameHeight);
     const endDepth = Math.ceil(viewportBottom / stackFrameHeight);
 
     const innerContainerWidth =
       containerWidth - TIMELINE_MARGIN_LEFT - TIMELINE_MARGIN_RIGHT;
+    const innerDevicePixelWidth = innerContainerWidth * devicePixelRatio;
```

Comment:
The naming of pixel is a bit inconsistent here and on other places. For example, on line 163, it is `devicePixelsWidth`, but here there's no plural `s`: `innerDevicePixelWidth`. Perhaps it's intentional, but i don't think it is?

Reply:
Yeah, that is really inconsistent. I'm going to make them all plural, since the types are plural. I'll keep the few places singular where they refer to a single pixel.